cmd/lotus-miner/actor.go

```diff
@@ -291,14 +294,21 @@ var actorWithdrawCmd = &cli.Command{
 			return err
 		}
 
-		var withdrawn abi.TokenAmount
-		if err := withdrawn.UnmarshalCBOR(bytes.NewReader(wait.Receipt.Return)); err != nil {
+		nv, err := api.StateNetworkVersion(ctx, wait.TipSet)
+		if err != nil {
 			return err
 		}
 
-		fmt.Printf("Successfully withdrew %s FIL\n", withdrawn)
-		if withdrawn != amount {
-			fmt.Printf("Note that this is less than the requested amount of %s FIL\n", amount)
+		if nv >= network.Version14 {
+			var withdrawn abi.TokenAmount
+			if err := withdrawn.UnmarshalCBOR(bytes.NewReader(wait.Receipt.Return)); err != nil {
+				return err
+			}
+
+			fmt.Printf("Successfully withdrew %s FIL\n", withdrawn)
+			if withdrawn != amount {
```

Comment:
In lotus - we are checking if the requested amount is > balance. if so, we will return an error directely 

https://github.com/filecoin-project/lotus/blob/1d2f84322d4fc1c0d5357476aa693c309097a8c8/cmd/lotus-miner/actor.go#L243-L259

so either we need to change that behavior or get rid of this check

Reply:
in actor - it does take the smaller value and process tho seems like 
https://github.com/filecoin-project/specs-actors/blob/855f60561dfc014fe7e601402de3621d3219df47/actors/builtin/miner/miner_actor.go#L2052

Reply:
Hmm, that's interesting...i'd prefer to leave both actually. 

I think if someone is explicitly trying to withdraw more than they can, we should fail and tell them. If a message winds up withdrawing less than expected, we should also yell at them about that. 

The latter case should only happen in unusual edge cases (the available balance changed between message signing and execution).